nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestCacheConversion.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.platform.convert.tests;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+import org.nuxeo.ecm.core.convert.cache.CacheKeyGenerator;
+import org.nuxeo.ecm.core.convert.cache.ConversionCacheHolder;
+
+/**
+ * @since 11.1
+ */
+public class TestCacheConversion extends BaseConverterTest {
+
+    @Test
+    public void shouldRegenNullFileCacheEntries() throws IOException {
+        String srcMT = "text/html";
+        String fileName = "hello.html";
+        String converterName = cs.getConverterName(srcMT, "application/pdf");
+        assertEquals("any2pdf", converterName);
+
+        checkConverterAvailability(converterName);
+        checkCommandAvailability("soffice");
+
+        BlobHolder hg = getBlobFromPath("test-docs/" + fileName, srcMT);
+        Map<String, Serializable> parameters = new HashMap<>();
+
+        // first conversion
+        BlobHolder result = cs.convert(converterName, hg, parameters);
+        assertNotNull(result);
+
+        // retrieve the temp file
+        String cacheKey = CacheKeyGenerator.computeKey(converterName, hg, parameters);
+        BlobHolder blobHolder = ConversionCacheHolder.getFromCache(cacheKey);
+        String firstTempFileName = blobHolder.getBlob().getFile().getAbsolutePath();
+        // delete the temp file
+        Files.delete(Paths.get(blobHolder.getBlob().getFile().getPath()));
+
+        // replay conversion
+        cs.convert(converterName, hg, parameters);
+        blobHolder = ConversionCacheHolder.getFromCache(cacheKey);
+        String secondTempFileName = blobHolder.getBlob().getFile().getAbsolutePath();
+        // missing temp file has been regenerated. So it has a different name
+        assertNotEquals(firstTempFileName, secondTempFileName);
+    }
+
+}
```

Comment:
This test is nice, but it might be better/simpler to have a real unit test of the impacted `ConversionCacheHolder` API:
- In the same Maven module: `nuxeo-core-convert` instead of `nuxeo-platform-convert`, so the test and the code are in the same layer.
- Testing directly `#getFromCache`, as it's static it should be easy.
- Not bothering with the ConversionService.

You should be able to add a `BlobHolder` to the cache, verify you can get it, delete its underlying file, get it again and verify the result is null and the cache entry has been removed.

Reply:
I changed the way of the test then.
Now I check after getting a null file cache entry, the cache clears the key rather than keeping it.

Reply:
Still don't understand we you need to go through the conversion service for this unit test...

Reply:
No more conversionService, all the calls are direct to `ConversionCacheHolder` static methods